### ISSUE TITLE
Make non-droppable area bigger around eye toggle

### ISSUE THREAD

**Issue by catherinecarter** [phetsims/number-pairs#38]
While interviewing, a student intentionally put the kittens "all over the place." The corners were their favorite location (farthest from each other possible), and therefore put two kittens over they eye toggle: 
<img width="114" alt="image" src="https://github.com/user-attachments/assets/3ee76cf0-83ca-495b-95cd-a8c54252ba4c">

Recommendation: to make objects repel away from the eye toggle so when the user touches something near it, the intended action is interpreted by the sim (e.g., if the user moves an object close by and subsequently wants to activate the eye toggle, they can without needing move objects out of the way).

**Comment by marlitas** [comment]
The main things that need to get addressed in this issue are:

- When an object is dropped (through mouse or keyboard) any objects that were behind it should move out of the way.
  - Objects that move out of the way should not move outside the bounds of the CountingAreaNode, and should not cover the eye toggle.
- When an object is dropped (through mouse of keyboard) on top of the eye toggle, it should move over so that it is not blocking it.
  - The final position of the dropped object should not be covering another object, and should be within the bounds of the CountingAreaNode.

Currently there is rudimentary logic for this in CountingAreaNode:

-  dropCountingObject
- getDropZoneBounds
- sendToValidDropPoint
- getValidDropPoint

**Comment by pixelzoom** [comment]
This seems like an unnecessary complication.  Have you considered putting the eye toggle buttons _outside_ the box? There is plenty of space below the box.

**Comment by pixelzoom** [comment]
I could also make an argument that putting the eye toggles outside the box is a better, more consistent UX.  Better because the student doesn't have to try to avoid dragging onto the toggles. Better because the eye toggle may be slightly more obvious outside the box.  And more consistent because other buttons that affect the contents of the box (organize, swap) are outside the box.

**Comment by marlitas** [comment]
I will let @catherinecarter answer this as I agree that putting it outside the counting area would simplify development.

**Comment by catherinecarter** [comment]
We could try that, I agree that it would make the code simpler. 

In terms of discoverability, however, I think putting the eye toggle below the countingArea makes it more confusing for the user to know what the toggle is controlling. And, it looks weird below the countingArea on the left.

<img width="550" alt="Image" src="https://github.com/user-attachments/assets/e2c107de-6c5a-4296-8291-7c8b22324fb0" />

In all other sims with an eye toggle, the toggle is to the right or left of the thing it controls (e.g., calculus grapher, graphing lines), except for function builder where the eye toggle is centered below the thing it controls. 

<img width="350" alt="Image" src="https://github.com/user-attachments/assets/0d345642-16d0-49d4-9a79-d97dbc961184" /> <img width="350" alt="Image" src="https://github.com/user-attachments/assets/7ebc085c-b836-4205-a5d7-f0d1c568ee92" /> 

<img width="350" alt="Image" src="https://github.com/user-attachments/assets/c22702a8-b09d-4124-98ac-7b3d2ff0cc0c" />


If we could put it to the right or left of the countingArea, it would make more sense, but there is not enough room for the eye toggle to be on the right of the location-based representation where there are two eye toggles, and I feel like it would be weird to put one toggle to the left and the other below (maybe, maybe not?), like shown below:
<img width="500" alt="Image" src="https://github.com/user-attachments/assets/1cea2cd5-01aa-43f0-9bcf-95d0938743cd" />

Since it makes more sense to not move left the eye toggle from screen to screen, we'd have to consider the toggle either under or to the left of the countingArea. Here's what it would look like to have the eye toggle below the countingArea in other representations (which I think looks awkward and unintuitive):

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/e31ffece-d08a-4749-9907-a8c23a6c287d" />

What makes more sense is to have the eye toggle to the left of the countingArea in many ways:
* functionally for what it controls, 
* visually to know what the toggle is controlling, and 
* consistently across sims.

<img width="500" alt="Image" src="https://github.com/user-attachments/assets/6adaa3d2-54ca-48d0-84c2-4052edd9e88d" />

Ideas?

**Comment by pixelzoom** [comment]
I agree that the toggle button looks better to the left of the countingArea and is more consistent with other sims. But I didn't suggest that because it's not possible in for the location-based representations, where the left and right halves of the countingArea each have their own toggle button.   And having the 2 toggle buttons in different (inconsistent) locations for left and right halves seems way worse than the compromise of putting them both below the countingArea.

**Comment by pixelzoom** [comment]
I said:
> ... There is plenty of space below the boxes, in all screens.

My bad. That's not the case for the Number Line:

<img width="356" alt="Image" src="https://github.com/user-attachments/assets/6d0f8eb4-b6ba-419d-ae4d-487d969aea20" />

I don't think it would be terrible to put the eye toggle button to the left of the countingArea for all cases where there is 1 such button.  And put the eyes toggle buttons below the counting area when there is a pair of buttons.

**Comment by ariel-phet** [comment]
Ok, I am not sure this avoids any complications, but a question...

What about if the eye toggle is tucked into the corner of the box, but the "box" does not actually include the eye toggle (so the area you can drag the kittens has that corner where the eye toggle is "cut out"). Then you could never drag the kittens over it in the first place. I definitely think any kind of place where we have movable objects and then some kind of repulsion when you let them go gets tricky and bug prone, so definitely see the value in not going that direction. But I feel like we have lots of examples of where a certain component cannot overlap other components. 

There is no value in having that button be covered so why not just make it impossible to cover it? Like:

<img width="961" alt="Image" src="https://github.com/user-attachments/assets/2046716c-6284-4a33-aada-3aca3356a562" />

**Comment by marlitas** [comment]
Unfortunately drag bounds are a rectangle, not a polygon with the possibility of multiple vertices. So we can make the overall drag bounds smaller (like perhaps nothing can drag below the topmost point of the eye toggle), but we cannot cut out a rectangle in the corner from the original bounds.

**Comment by pixelzoom** [comment]
Non-rectangular drag bounds can be done.  But there is no scenery or common-code support. It was attempted in another sim that I reviewed not too long ago (My Solar System? Kepler's Laws?) and it was a pain point -- complicated and  new problems seemed to be created whenever the code was touched.  I don't recommend it.

**Comment by marlitas** [comment]
> Non-rectangular drag bounds can be done. But there is no scenery or common-code support.

Oh that's good to know. I just assumed because there wasn't Scenery support for it that it was not possible or at least not possible for PhET sims.

**Comment by ariel-phet** [comment]
@pixelzoom @marlitas @catherinecarter to me, shrinking the drag bounds on the bottom is not a bad way to go. The kittens already have a bit of padding in terms of the top drag bound because of the toggle switch. I don't think it will feel too odd or unsymmetric to make the bottom drag bound different from the visual play area. I think that is a reasonable trade off for simplicity and won't look odd if you put some kittens to there top max position and bottom max position (it will actually feel fairly even).

**Comment by catherinecarter** [comment]
After consulting with @amanda-phet and @ariel-phet, we all agreed that allowing students to occlude the eye toggle isn't a huge deal since you can still see at least part of the toggle at all times since the objects themselves repel each other.

Decision: Leave the eye toggle inside the `countingArea` for all representations for consistency of location and ease of connecting the function of the eye toggle to the location of it.

**Comment by marlitas** [comment]
Great! I refactored the code it is now much cleaner and you can't push things outside of the counting area.
@catherinecarter this is ready for review. Close if all looks well.

**Comment by marlitas** [comment]
Ahh! Nevermind one sec to fix a silly mistake.

**Comment by marlitas** [comment]
Fixed! now you can review @catherinecarter :-)

**Comment by catherinecarter** [comment]
Looks good. I couldn't get any kittens to fall out of the play area.

**Comment by marlitas** [comment]
Great.  I believe this can be closed. Thanks!
